idd/Energy+.idd.in

```diff
@@ -91303,809 +91272,129 @@ Curve:ChillerPartLoadWithLift,
-  A1 , \field Name
+Table:IndependentVariable,
+       \memo An independent variable representing a single dimension of a Table:Lookup
+       \memo object.
+       \extensible:1
+  A1,  \field Name
        \required-field
+       \reference IndependentVariableName
        \type alpha
-       \reference UnivariateFunctions
-  A2 , \field Curve Type
+  A2,  \field Interpolation Method
        \type choice
        \key Linear
-       \key Quadratic
        \key Cubic
-       \key Quartic
-  A3 , \field Interpolation Method
+       \default Linear
+  A3,  \field Extrapolation Method
        \type choice
-       \key LinearInterpolationOfTable
-       \key EvaluateCurveToLimits
-       \key LagrangeInterpolationLinearExtrapolation
-  N1 , \field Minimum Value of X
+       \key Constant
+       \key Linear
+       !\key Unavailable
+       \default Constant
+  N1,  \field Minimum Value
        \type real
        \unitsBasedOnField A4
-       \note used only when Interpolation Type is Evaluate Curve
-       \note to Limits
-  N2 , \field Maximum Value of X
+  N2,  \field Maximum Value
        \type real
        \unitsBasedOnField A4
-       \note used only when Interpolation Type is Evaluate Curve
-       \note to Limits
-  N3 , \field Minimum Table Output
-       \type real
-       \note Specify the minimum value calculated by this table
-       \note lookup object
-       \unitsBasedOnField A5
-       \note used only when Interpolation Type is Evaluate Curve
-       \note to Limits
-  N4 , \field Maximum Table Output
+  N3,  \field Normalization Value
        \type real
-       \note Specify the maximum value calculated by this table
-       \note lookup object
-       \unitsBasedOnField A5
-       \note used only when Interpolation Type is Evaluate Curve
-       \note to Limits
-  A4,  \field Input Unit Type for X
+       \unitsBasedOnField A4
+  A4,  \field Unit Type
        \type choice
        \key Dimensionless
        \key Temperature
        \key VolumetricFlow
        \key MassFlow
        \key Power
        \key Distance
+       \key Angle
        \default Dimensionless
-  A5,  \field Output Unit Type
-       \type choice
-       \key Dimensionless
-       \key Capacity
-       \key Power
-       \key Temperature
-       \default Dimensionless
-  N5 , \field Normalization Reference
-       \type real
-       \note This field is used to normalize the following ouput data.
-       \note The minimum and maximum table output fields are also normalized.
-       \note If this field is blank or 1, the table data presented
-       \note in the following fields will be used with normalization
-       \note reference set to 1.
-  N6 , \field X Value #1
-       \required-field
+  A5,  \field External File Name
+       \type alpha
+       \retaincase
+  N4,  \field External File Column Number
+       \type integer
+       \minimum 1
+  N5,  \field External File Starting Row Number
+       \type integer
+       \minimum 1
+  N6;  \field Value 1
```

Comment:
All of the new table object stop with only one instance of the extensible fields. This is going to prevent opening any file with table objects in IDF Editor. At a minimum, it seems that Table:IndependentVariableList should be expanded here to, say 10 variables so at least an idf that uses an external file can be opened. Whether we extend the Table:Lookup Output Value N fields here is debatable. If we want to be able to open the UnitaryHybrid example files, then 1500 or so should do it.

Reply:
I thought IDF editor was modified to do this automatically. Maybe that was 2.0?

Reply:
IDF Editor 2.0 is not finished yet.